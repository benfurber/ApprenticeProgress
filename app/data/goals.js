const goalsData = {
  [1]: {
    actions: ["1", "2"],
    description: "HELLO",
    score: 1,
    tag: "Team thoughtbot",
    title: "I contribute to thoughtbot being a good place to work",
  },
  [2]: {
    actions: ["1", "2"],
    description: "",
    score: 2,
    tag: "Team thoughtbot",
    title: "I iterate over my existing knowledge effectively",
  },
  [3]: {
    actions: ["1", "2"],
    description: "",
    score: 3,
    tag: "Team thoughtbot",
    title: "I’m a great pair programmer",
  },
  [4]: {
    actions: ["1", "2"],
    description: "",
    score: 1,
    tag: "Team thoughtbot",
    title: "I help others become better ",
  },
  [5]: {
    actions: ["1", "2"],
    description: "",
    score: 3,
    tag: "Team thoughtbot",
    title: "I’m passionate about software/products",
  },
  [6]: {
    actions: ["1", "2"],
    description: "",
    score: 4,
    tag: "Team thoughtbot",
    title: "I set my own direction",
  },
};

export { goalsData };
