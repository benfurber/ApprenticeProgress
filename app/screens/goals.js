import React, { Component } from "react";
import { Container } from "native-base";
import { Goal } from "components";

export class Goals extends Component {
  getGoals(goalsData) {
    return goalsData.goals.map((goal, index) => {
      return this.displayGoal(goal, index);
    });
  }

  displayGoal(goal, index) {
    return <Goal data={ goal } key={ index } />;
  }

  render() {
    return (
      <Container>
        { this.getGoals(goalsData) }
      </Container>
    );
  }
}

const goalsData = {
  goals: [
    {
      "title": "Goal 1",
      "description": "Desc",
    },
    {
      "title": "Goal 2",
      "description": "Desc",
    },
  ],
};
