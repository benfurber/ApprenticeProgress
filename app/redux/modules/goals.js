import { goalsData } from "../../data/goals";

const GET = "goals/get";

const goalsAction = () => ({ type: GET });

const goalsReducer = (state, action) => {
  switch (action.type) {
    case GET:
      return { ...state, goals: goalsData };
    default:
      return state;
  }
};

export { goalsAction, goalsReducer };