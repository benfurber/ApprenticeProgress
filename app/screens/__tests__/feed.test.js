// @flow

import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import { Feed } from "../Feed";
import navigationMock from "mocks";

import { goalsData } from "../../data/goals";

const middlewares = [];
const mockStore = configureStore(middlewares);

describe("Feed", () => {
  it("renders correctly", () => {
    const store = mockStore({
      goals: goalsData,
    });

    const component = shallow(<Feed navigation={navigationMock} store={store} />);

    expect(component).toMatchSnapshot();
  });
});
