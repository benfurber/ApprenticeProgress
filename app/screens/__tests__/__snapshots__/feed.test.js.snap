// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Feed)
    navigation={
      Object {
        "getParam": [Function],
        "navigate": [MockFunction],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "goalsAction": Object {
        "type": "goals/get",
      },
      "navigation": Object {
        "getParam": [Function],
        "navigate": [MockFunction],
      },
      "state": Object {
        "goals": Object {
          "1": Object {
            "actions": Array [
              "Help make the office a great place to work",
              "Pick up tasks others normally do when appropriate",
            ],
            "description": "thoughtbot must be a great place to work for all staff.",
            "evidence": Array [
              "Sorting snacks, etc. for office.",
            ],
            "score": 4,
            "tag": "Team thoughtbot",
            "title": "I contribute to thoughtbot being a good place to work",
          },
          "10": Object {
            "actions": Array [],
            "description": "",
            "evidence": Array [
              "Can explain the basics to non-developers",
            ],
            "score": 3,
            "tag": "I'm a developer",
            "title": "I understand how React Native works",
          },
          "11": Object {
            "actions": Array [
              "Build an API",
              "Deploy an app to the marketplace",
            ],
            "description": "Fundamentals of the how the internet is possible.",
            "evidence": Array [
              "",
            ],
            "score": 3,
            "tag": "I'm a developer",
            "title": "I understand how the web/databases works",
          },
          "12": Object {
            "actions": Array [
              "Example of difficult conversations",
            ],
            "description": "Central to how thoughtbot work.",
            "evidence": Array [
              "Good day-to-day communication with first client",
            ],
            "score": 3,
            "tag": "I'm a consultant",
            "title": "I communicate clearly with clients",
          },
          "13": Object {
            "actions": Array [
              "Participate in a sales meeting",
            ],
            "description": "All thoughtbot staff contribute to the sales process.",
            "evidence": Array [
              "Could give boilerplate thoughtbot explanation",
            ],
            "score": 2,
            "tag": "I'm a consultant",
            "title": "I can clearly explain to others where/when thoughtbot can add value",
          },
          "14": Object {
            "actions": Array [],
            "description": "All thoughtbot developers aspire to be CTOs for clients.",
            "evidence": Array [
              "Have shown eye for detail in commits, communication and meetings",
            ],
            "score": 3,
            "tag": "I'm a consultant",
            "title": "I understand how technical/design decisions will impact clients",
          },
          "15": Object {
            "actions": Array [],
            "description": "thoughtbot build software clients and users want.",
            "evidence": Array [
              "First client continue to be happy with output",
            ],
            "score": 0,
            "tag": "I'm a consultant",
            "title": "My teams build software clients’ love",
          },
          "2": Object {
            "actions": Array [
              "Continue to learn React Native",
              "Pick up client projects quickly",
              "Solve own bugs/technical challenges without needing support",
            ],
            "description": "I can support my own learning.",
            "evidence": Array [
              "Completed a 5 point story with only gentle support.",
            ],
            "score": 3,
            "tag": "Team thoughtbot",
            "title": "I iterate over my existing knowledge effectively",
          },
          "3": Object {
            "actions": Array [],
            "description": "Collaborating with colleagues is essential to innovation.",
            "evidence": Array [
              "Refreshing the pairing programming skills of pairs",
            ],
            "score": 4,
            "tag": "Team thoughtbot",
            "title": "I’m a great pair programmer",
          },
          "4": Object {
            "actions": Array [
              "Run workshop for team to discuss effective pairing methods",
            ],
            "description": "Everyone in thoughtbot helps each other learn and grow",
            "evidence": Array [
              "Refreshing the pairing programming skills of pairs",
            ],
            "score": 2,
            "tag": "Team thoughtbot",
            "title": "I help others become better",
          },
          "5": Object {
            "actions": Array [
              "1",
              "2",
            ],
            "description": "",
            "evidence": Array [
              "Eye for detail",
              "Wanting to make products better (not just story minimum)",
            ],
            "score": 3,
            "tag": "Team thoughtbot",
            "title": "I’m passionate about software/products",
          },
          "6": Object {
            "actions": Array [],
            "description": "Self management requires that you can motivate yourself to do better.",
            "evidence": Array [
              "",
            ],
            "score": 4,
            "tag": "Team thoughtbot",
            "title": "I set my own direction",
          },
          "7": Object {
            "actions": Array [
              "Implement redux",
            ],
            "description": "Code is object orientated, easy to read/understand and easy to change.",
            "evidence": Array [
              "Can explain difference and value of different types of components.",
              "Came up with the idea of the internal link component",
            ],
            "score": 2.5,
            "tag": "I'm a developer",
            "title": "I can write high quality code for React Native",
          },
          "8": Object {
            "actions": Array [
              "Continue to write more complex tests",
              "Write unit tests for redux",
            ],
            "description": "Test driven development is central to how thoughtbot work.",
            "evidence": Array [
              "Over initial hurdle of basics",
            ],
            "score": 2.5,
            "tag": "I'm a developer",
            "title": "I can write high quality test suites for React Native",
          },
          "9": Object {
            "actions": Array [
              "Contribute to client projects",
              "Build breakable toy",
            ],
            "description": "Bit difficult to do job otherwise...",
            "evidence": Array [
              "Completed a few screens/components",
            ],
            "score": 2.5,
            "tag": "I'm a developer",
            "title": "I can build apps in React Native",
          },
        },
      },
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "goalsAction": Object {
          "type": "goals/get",
        },
        "navigation": Object {
          "getParam": [Function],
          "navigate": [MockFunction],
        },
        "state": Object {
          "goals": Object {
            "1": Object {
              "actions": Array [
                "Help make the office a great place to work",
                "Pick up tasks others normally do when appropriate",
              ],
              "description": "thoughtbot must be a great place to work for all staff.",
              "evidence": Array [
                "Sorting snacks, etc. for office.",
              ],
              "score": 4,
              "tag": "Team thoughtbot",
              "title": "I contribute to thoughtbot being a good place to work",
            },
            "10": Object {
              "actions": Array [],
              "description": "",
              "evidence": Array [
                "Can explain the basics to non-developers",
              ],
              "score": 3,
              "tag": "I'm a developer",
              "title": "I understand how React Native works",
            },
            "11": Object {
              "actions": Array [
                "Build an API",
                "Deploy an app to the marketplace",
              ],
              "description": "Fundamentals of the how the internet is possible.",
              "evidence": Array [
                "",
              ],
              "score": 3,
              "tag": "I'm a developer",
              "title": "I understand how the web/databases works",
            },
            "12": Object {
              "actions": Array [
                "Example of difficult conversations",
              ],
              "description": "Central to how thoughtbot work.",
              "evidence": Array [
                "Good day-to-day communication with first client",
              ],
              "score": 3,
              "tag": "I'm a consultant",
              "title": "I communicate clearly with clients",
            },
            "13": Object {
              "actions": Array [
                "Participate in a sales meeting",
              ],
              "description": "All thoughtbot staff contribute to the sales process.",
              "evidence": Array [
                "Could give boilerplate thoughtbot explanation",
              ],
              "score": 2,
              "tag": "I'm a consultant",
              "title": "I can clearly explain to others where/when thoughtbot can add value",
            },
            "14": Object {
              "actions": Array [],
              "description": "All thoughtbot developers aspire to be CTOs for clients.",
              "evidence": Array [
                "Have shown eye for detail in commits, communication and meetings",
              ],
              "score": 3,
              "tag": "I'm a consultant",
              "title": "I understand how technical/design decisions will impact clients",
            },
            "15": Object {
              "actions": Array [],
              "description": "thoughtbot build software clients and users want.",
              "evidence": Array [
                "First client continue to be happy with output",
              ],
              "score": 0,
              "tag": "I'm a consultant",
              "title": "My teams build software clients’ love",
            },
            "2": Object {
              "actions": Array [
                "Continue to learn React Native",
                "Pick up client projects quickly",
                "Solve own bugs/technical challenges without needing support",
              ],
              "description": "I can support my own learning.",
              "evidence": Array [
                "Completed a 5 point story with only gentle support.",
              ],
              "score": 3,
              "tag": "Team thoughtbot",
              "title": "I iterate over my existing knowledge effectively",
            },
            "3": Object {
              "actions": Array [],
              "description": "Collaborating with colleagues is essential to innovation.",
              "evidence": Array [
                "Refreshing the pairing programming skills of pairs",
              ],
              "score": 4,
              "tag": "Team thoughtbot",
              "title": "I’m a great pair programmer",
            },
            "4": Object {
              "actions": Array [
                "Run workshop for team to discuss effective pairing methods",
              ],
              "description": "Everyone in thoughtbot helps each other learn and grow",
              "evidence": Array [
                "Refreshing the pairing programming skills of pairs",
              ],
              "score": 2,
              "tag": "Team thoughtbot",
              "title": "I help others become better",
            },
            "5": Object {
              "actions": Array [
                "1",
                "2",
              ],
              "description": "",
              "evidence": Array [
                "Eye for detail",
                "Wanting to make products better (not just story minimum)",
              ],
              "score": 3,
              "tag": "Team thoughtbot",
              "title": "I’m passionate about software/products",
            },
            "6": Object {
              "actions": Array [],
              "description": "Self management requires that you can motivate yourself to do better.",
              "evidence": Array [
                "",
              ],
              "score": 4,
              "tag": "Team thoughtbot",
              "title": "I set my own direction",
            },
            "7": Object {
              "actions": Array [
                "Implement redux",
              ],
              "description": "Code is object orientated, easy to read/understand and easy to change.",
              "evidence": Array [
                "Can explain difference and value of different types of components.",
                "Came up with the idea of the internal link component",
              ],
              "score": 2.5,
              "tag": "I'm a developer",
              "title": "I can write high quality code for React Native",
            },
            "8": Object {
              "actions": Array [
                "Continue to write more complex tests",
                "Write unit tests for redux",
              ],
              "description": "Test driven development is central to how thoughtbot work.",
              "evidence": Array [
                "Over initial hurdle of basics",
              ],
              "score": 2.5,
              "tag": "I'm a developer",
              "title": "I can write high quality test suites for React Native",
            },
            "9": Object {
              "actions": Array [
                "Contribute to client projects",
                "Build breakable toy",
              ],
              "description": "Bit difficult to do job otherwise...",
              "evidence": Array [
                "Completed a few screens/components",
              ],
              "score": 2.5,
              "tag": "I'm a developer",
              "title": "I can build apps in React Native",
            },
          },
        },
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
